core/embed/rust/src/ui/model_tt/layout.rs

```diff
@@ -338,17 +342,72 @@ extern "C" fn new_confirm_blob(n_args: usize, args: *const Obj, kwargs: *mut Map
     unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
 }
 
+extern "C" fn new_confirm_properties(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    let block = move |_args: &[Obj], kwargs: &Map| {
+        let title: StrBuffer = kwargs.get(Qstr::MP_QSTR_title)?.try_into()?;
+        let hold: bool = kwargs.get_or(Qstr::MP_QSTR_hold, false)?;
+        let items: Obj = kwargs.get(Qstr::MP_QSTR_items)?;
+
+        let mut paragraphs = ParagraphVecLong::new();
+
+        let mut iter_buf = IterBuf::new();
+        let iter = Iter::try_from_obj_with_buf(items, &mut iter_buf)?;
+        for para in iter {
+            let [key, value, is_mono]: [Obj; 3] = iter_into_objs(para)?;
+            let key = key.try_into_option::<StrBuffer>()?;
+            let value = value.try_into_option::<StrBuffer>()?;
+
+            if let Some(key) = key {
+                if value.is_some() {
+                    paragraphs.add(Paragraph::new(&theme::TEXT_BOLD, key).no_break());
+                } else {
+                    paragraphs.add(Paragraph::new(&theme::TEXT_BOLD, key));
+                }
+            }
+            if let Some(value) = value {
+                if is_mono.try_into()? {
+                    paragraphs.add(Paragraph::new(&theme::TEXT_MONO, value));
+                } else {
+                    paragraphs.add(Paragraph::new(&theme::TEXT_NORMAL, value));
+                }
```

Comment:
sidenote: we should violently throw out this flexibility of sometimes specifying only key and sometimes specifying only value. I believe only Cardano is using that, maybe a little bit of UX work is needed on that side.